tslearn/clustering.py

```diff
@@ -30,6 +33,99 @@
 # Mathieu Blondel, under BSD 3 clause license
 
 
+def _k_init_metric(X, n_clusters, cdist_metric, random_state,
+                   n_local_trials=None):
+    """Init n_clusters seeds according to k-means++ with a custom distance
+    metric.
+
+    Parameters
+    ----------
+    X : array or sparse matrix, shape (n_samples, n_features)
+        The data to pick seeds for. To avoid memory copy, the input data
+        should be double precision (dtype=np.float64).
+
+    n_clusters : integer
+        The number of seeds to choose
+
+    cdist_metric : function
+        Function to be called for cross-distance computations
+
+    random_state : RandomState instance
+        Generator used to initialize the centers.
+
+    n_local_trials : integer, optional
+        The number of seeding trials for each center (except the first),
+        of which the one reducing inertia the most is greedily chosen.
+        Set to None to make the number of trials depend logarithmically
+        on the number of seeds (2+log(k)); this is the default.
```

Comment:
Number of clusters?

Reply:
Nope, it's something in the algo, at each step, you pick a certain number of candidates at random (with probability proportional to their squared distance to the closest center in the current set) and you keep the one that minimizes the sum of distances to other training samples.